website/docs/language/modules/develop/refactoring.html.md

```diff
@@ -0,0 +1,475 @@
+(a `data` block).
+
+There is one exception to that constraint: a provider developer may mark a
+particular resource type name as deprecated and nominate a new name to replace
+it. In that case, the provider developer is responsible for making the old
+and new names schema-compatible, and can then tell Terraform that it's safe to
+support changing from the old to the new type via a `moved` block. Provider
+documentation will include an example for any situation where this exception
+applies.
+
+## Enabling `count` or `for_each` For a Resource
+
+Imagine that the initial version of your module contains a single-instance
+resource:
+
+```hcl
+resource "aws_instance" "a" {
+  # (resource-type-specific configuration)
+}
+```
+
+Applying this configuration would lead to Terraform creating an object
+bound to the address `aws_instance.a`.
+
+Later requirements might lead to you needing to use
+[`for_each`](../meta-arguments/for_each.html) with this resource, in order
+to systematically declare multiple instances. To preserve an object that
+was previously associated with `aws_instance.a` alone, you must used a `moved`
+block to specify which instance key that object will take in the new
+configuration:
+
+```hcl
+locals {
+  instances = tomap({
+    big = {
+      instance_type = "m3.large"
+    }
+    small = {
+      instance_type = "t2.medium"
+    }
+  })
+}
+
+resource "aws_instance" "a" {
+  for_each = local.instances
+
+  instance_type = each.value.instance_type
+  # (other resource-type-specific configuration)
+}
+
+moved {
+  from = aws_instance.a
+  to   = aws_instance.a["small"]
+}
+```
+
+The above will avoid Terraform planning to destroy any existing object at
+`aws_instance.a`, treating it instead as if it were originally created
+as `aws_instance.a["small"]`.
+
+When at least one of the two addresses includes an instance key, like
+`["small"]` in the above example, Terraform will understand both addresses
+as referring to specific _instances_ of a resource rather than the resource
+as a whole. That means you can use `moved` to switch between keys and to
+add and remove keys as you switch between `count`, `for_each`, or neither.
+
+The following are some other examples of valid `moved` blocks that record
+changes to resource instance keys in a similar way:
+
+```hcl
+# Both old and new configuration used "for_each", but the
+# "small" element was renamed to "tiny".
+moved {
+  from = aws_instance.b["small"]
+  to   = aws_instance.b["tiny"]
+}
+
+# The old configuration used "count" and the new configuration
+# uses "for_each", with the following mappings from
+# index to key:
+moved {
+  from = aws_instance.c[0]
+  to   = aws_instance.c["small"]
+}
+moved {
+  from = aws_instance.c[1]
+  to   = aws_instance.c["tiny"]
+}
+
+# The old configuration used "count" and the new configuration
+# uses either "count" nor "for_each", and we want to keep
+# only the object at index 2.
```

Comment:
should you change it to "or" rather than "nor"? OR should be be neither? 

Also, is there a way to rephrase without saying "we". Maybe we just change to "you"

Reply:
Hmm yes, I intended this to be "neither". Thanks!

Regarding the "we"/"you" thing, I think historically we've typically used "we" in situations where we're leading someone through a hypothetical example, and "you" for when we're talking about something the reader _actually_ wants, though I'm probably overstating how consistent we've been in that way.

For this particular situation it feels weird to me to _tell_ the reader what they want, but maybe that's okay. Another option that I think of considering your feedback here would be to phrase it as an "If you want ... then ..." sentence, perhaps like this:

```hcl
# The old configuration used "count" and the new configuration
# uses either "count" nor "for_each". If you want to keep the
# object at index 2, declare that it has moved to the new
# address.
moved {
  from = aws_instance.d[2]
  to   = aws_instance.d
}
```

Does that seem reasonable?

(I guess I'd also need to similarly edit the other examples above, because they are all currently phrased as descriptions of a situation where the example would apply, rather than as an instruction for the reader to follow, but I'll wait until I hear what you think about _this_ example before I attempt that! :grinning: )